src/geometry/PolySet.h

```diff
@@ -8,12 +8,20 @@
 
 #include <vector>
 #include <string>
+#include <unordered_map>
+#include <hash.h>
+
+int operator==(const Vector3d &a, const Vector3d &b);
 
 class PolySet : public Geometry
 {
 public:
   VISITABLE_GEOMETRY();
-  Polygons polygons;
+  PolygonsInd polygons_ind;
```

Comment:
When reading this, I wonder if an indexed polygon mesh should ideally differ from an indexed triangle mesh. For rendering, we're going to have to convert to triangle meshes anyway, so perhaps it would be smart and efficient to make that choice earlier.